cmd/skaffold/app/cmd/deploy.go

```diff
@@ -30,11 +33,33 @@ func NewCmdDeploy(out io.Writer) *cobra.Command {
 		Args:  cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			// Same actions as `skaffold run`, but with pre-built images.
-			return run(out)
+			opts.Command = "deploy"
+			return runDeploy(out)
 		},
 	}
 	AddRunDevFlags(cmd)
 	AddRunDeployFlags(cmd)
 	cmd.Flags().StringSliceVar(&opts.PreBuiltImages, "images", nil, "A list of pre-built images to deploy")
 	return cmd
 }
+
+func runDeploy(out io.Writer) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	catchCtrlC(cancel)
+	runner, _, err := newRunner(opts)
+	if err != nil {
+		return errors.Wrap(err, "creating runner")
+	}
+	defer runner.RPCServerShutdown()
+
+	// TODO: This should be fixed with https://github.com/GoogleContainerTools/skaffold/issues/922#issuecomment-474710200
+	// where buildArtifacts will consume `skaffold build` output
+	buildArtifacts := []build.Artifact{}
+	if err := runner.Deploy(ctx, out, buildArtifacts); err != nil {
+		return err
+	}
+
+	runner.TailLogs(ctx, out, nil, buildArtifacts)
```

Comment:
The error is not handled here.

Also, the `TailLogs` function was unused before this PR. `runner.Run()` uses its own code.
I think we should make sure both `Run` and `Deploy` use the same code.

Reply:
i can look in to this.